### DISCUSSION TITLE
New feature request: stats while charging

### DISCUSSION THREAD

**Discussion by Spera15** [teslamate-org/teslamate#1849] [Ideas]
First of all thank you for this amazing project! It is really well done and helpful!

**Request - Charge monitor**
Using TM I can basically track every single aspect of my Tesla, except a single one: the charging process. There is a huge amount of stats related to completed charges (and it is probably the page I use the most), but at the moment there is no way to track a charge while it is being done.

It would be great if a new page could be added, that includes:
- a chart like the one in "overview", showing the undergoing charge
- battery level
- charging voltage
- charging kW
- charging Amps
- kWh added
- elapsed time
- estimated time left (if provided through Tesla API, I don't know if it does)
- estimated cost (if geofencing is available)
- estimated range
- estimated range added
- any other applicable stat

**Comment by adriankumpf** [comment]
Feature requests belong here: [💡 Ideas](https://github.com/adriankumpf/teslamate/discussions/categories/ideas). I'll move it.

**Comment by mat1990dj** [comment]
Also add charge rate, this is more precise for low power chargers as it reports km added per hour of charge instantly
https://github.com/adriankumpf/teslamate/discussions/2280

**Comment by DumSkidderik** [comment]
You can get the current charging session under timeline, then click the current parking / idle session and scroll down to the bottom, there will be a charging session related to the parking session, that's the current charging session.
Not sure if it's an oversight?
